app/models/manageiq/providers/nuage/network_manager/vsd_client.rb

```diff
@@ -93,6 +93,13 @@ def exclude_name(name)
 
     def get_list(url)
       response = @rest_call.get("#{@server}/#{url}")
+
+      # Reset headers after call was performed or next API call will send them too.
+      @rest_call.reset_headers
+
+      # API returns 204 No Content if too strict filter is defined.
+      return [] if response.code == 204
```

Comment:
May I suggest you merge this with L104, i.e.

```
return [] if response.body.empty? or response.code == 204
```

Reply:
Good point. I actually modified behavior to always return empty list `[]` unless status is 200 and body is present. Not sure why we complicated in first place.